packages/shared-ux/page/solution_nav/src/with_solution_nav.tsx

```diff
@@ -8,31 +8,40 @@
 
 import React, { ComponentType, useState } from 'react';
 import classNames from 'classnames';
-import { useIsWithinBreakpoints } from '@elastic/eui';
-import { EuiPageSideBarProps } from '@elastic/eui/src/components/page/page_side_bar';
-import { KibanaPageTemplateSolutionNav, KibanaPageTemplateSolutionNavProps } from './solution_nav';
-import { KibanaPageTemplateProps } from './types';
+import { useIsWithinBreakpoints, EuiPageTemplateProps } from '@elastic/eui';
+import { SolutionNav, SolutionNavProps } from './solution_nav';
 
 // https://reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging
 function getDisplayName(Component: ComponentType<any>) {
   return Component.displayName || Component.name || 'UnnamedComponent';
 }
 
-type SolutionNavProps = KibanaPageTemplateProps & {
-  solutionNav: KibanaPageTemplateSolutionNavProps;
+type TemplateProps = Pick<
+  EuiPageTemplateProps,
+  'pageSideBar' | 'pageSideBarProps' | 'template' | 'children'
+>;
```

Comment:
Why is this type so restrictive of EuiPageTemplateProps? Shouldn't It should extend them all as it does in `KibanaPageTemplateProps`/

Reply:
These are the props that this HOC requires.  It doesn't prevent the wrapped component from using the full extent of `KibanaTemplateProps`, but it specifically outlines those that this HOC expects and will use.